nexus/src/db/datastore.rs

```diff
@@ -1376,13 +1376,23 @@ impl DataStore {
 
     /// See `disk_update_runtime()`.  This version should only be used from
     /// sagas, which do not currently have authn contexts.
-    // TODO-security Like project_delete_disk_no_auth(), this should be removed
-    // once we have support for authz checks from sagas.
-    pub async fn disk_update_runtime_no_auth(
+    pub async fn disk_update_runtime(
         &self,
+        opctx: &OpContext,
         authz_disk: &authz::Disk,
         new_runtime: &DiskRuntimeState,
     ) -> Result<bool, Error> {
+        // TODO-security This permission might be overloaded here.  The way disk
+        // runtime updates work is that the caller in Nexus first updates the
+        // Sled Agent to make a change, then updates to the database to reflect
+        // that change.  So by the time we get here, we better have already done
+        // an authz check, or we will have already made some unauthorized change
```

Comment:
Validating my understanding - *are* we doing an authz check before we send the modification request to the Sled Agent today?

I thought our implementation of authz was entirely DB-centric

Reply:
We do for Disks.  In "main":
https://github.com/oxidecomputer/omicron/blob/f615ee43803902fc8ed37a7e66a39677f885dbdb/nexus/src/nexus.rs#L1479

I haven't gotten to Instances yet, but we will do it there too.

> I thought our implementation of authz was entirely DB-centric

Yeah, I've really emphasized the database up to this point.  But we want to do authz checks before we do anything that might modify the state of the system (the _whole_ system -- including the state on all the Sleds and Sidecars).